File: Week_6-Python/sentimental-readability/readability.py
from cs50 import get_string


def get_gradelevel(S, L):
    gradelevel = 0.0588 * L - 0.296 * S - 15.8
    if gradelevel < 1:
        print("Before Grade 1")
    elif gradelevel > 1 and gradelevel < 16:
        print(f"Grade {round(gradelevel)}")
    else:
        print("Grade 16+")


def get_SL(text):
    L = 0
    S = 0
    for wordcount, word in enumerate(text.split(), start=1):
        for letter in word:
            if letter in {".", "!", "?"}:
                S += 1
            elif letter.isalpha():
                L += 1

    S = (100 / wordcount) * S
    L = (100 / wordcount) * L
    return S, L


def main():
    text = get_string("Text to test: ")
    S, L = get_SL(text)
    get_gradelevel(S, L)

    
main()

File: Week_2-Arrays/pset2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

float get_index(float words, float characters, float sentences)
{
    float L = (100 / words) * characters;
    float S = (100 / words) * sentences;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    return index;
}

int main(void)
{
    string passage = get_string("Text: \n");
    float sentencecount = 0.0;
    float wordcount = 1.0;
    float charcount = 0.0;
    for (int i = 0; i < strlen(passage); i++)
    {
        if (passage[i] == ' ')
        {
            wordcount++;
        }
        else if (passage[i] == '!' || passage[i] == '?' || passage[i] == '.')
        {
            sentencecount++;
        }
        else if (isalpha(passage[i]))
        {
            charcount++;
        }
    }

    float index = get_index(wordcount, charcount, sentencecount) + 0.5;
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int)index);
    }
}